core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/data/AccessTokenDataBinderImpl.java

```diff
@@ -135,11 +135,7 @@
                 accessToken.setAuthorities(authorities);
             }
 
-            accessTokenDAO.save(accessToken);
-        }
-
-        if (replaceExisting && existing != null) {
-            accessTokenDAO.delete(existing);
+            accessTokenDAO.merge(accessToken);
```

Comment:
what about the `replaceExistingFlag`? You just removed it!

Reply:
yes, merge function simply ensure the uniqueness constraint by checking if the token exists and replace it if so. otherwise, it will create a new register. so no need to perform this check here

Reply:
`replaceExisting` is an argument of `AccesstTokenDataBinderImpl#create` that your change would simply ignore.
Not good: it must be checked.

Reply:
It is not completely ignored. it is checked by the second if condition as before

Reply:
So, the current implementation will replace the token only if `replaceExisting` is true; with your change, the token will be replaced anyway, because `JPAAccessTokenDAO#merge` will do

```
if (existing != null) {
            entityManager().remove(existing);
        }
```
while the current code does
```
        if (replaceExisting && existing != null) {
            accessTokenDAO.delete(existing);
        }
```

Reply:
exactly. this is to assure that there can be only one access token associated with a particular user at an any given time and also a workaround to overcome thread not safe problem without affecting the performance(without using serialized isolation)

Reply:
...but there might be cases where such constraint should not apply (see the `@Column` annotation we've been discussing).

Reply:
To me, it should be enough to pass the `replaceExisting` flag to the new `AccessTokenDAO#replace` method, and behave accordingly.